File: sdk/nodejs/provider.ts
import * as pulumi from '@pulumi/pulumi'

/**
 * Provides an OpenFaaS Provider resource.
 */
export class Provider extends pulumi.ProviderResource {
  /**
   * Create a provider resource with the given unique name, arguments, and options.
   *
   * @param name The _unique_ name of the resource.
   * @param args The arguments to use to populate this provider's configuration.
   * @param opts A bag of options that control this resource's behavior.
   */
  constructor(name: string, args: ProviderArgs, opts?: pulumi.ResourceOptions) {
    super(
      'openfaas',
      name,
      {
        endpoint: args.endpoint,
        username: args.username,
        password: args.password,
        tlsSkipVerify: args.tlsSkipVerify,
      },
      opts
    )
  }
}

/**
 * The set of arguments for constructing a Provider resource.
 */
export interface ProviderArgs {
  readonly endpoint: pulumi.Input<string>
  readonly username?: pulumi.Input<string>
  readonly password?: pulumi.Input<string>
  readonly tlsSkipVerify?: pulumi.Input<boolean>
}

File: sdk/dotnet/Provider.cs
// Pulumi Command Provider .NET SDK
// Copyright 2020, Mitchell Maler.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

using System.Collections.Generic;
using System.Collections.Immutable;
using System.Threading.Tasks;
using Pulumi.Serialization;

namespace Pulumi.Command
{
    /// <summary>
    /// The provider type for the command package. By default, resources use package-wide configuration
    /// settings, however an explicit `Provider` instance may be created and passed during resource
    /// construction to achieve fine-grained programmatic control over provider settings. See the
    /// [documentation](https://www.pulumi.com/docs/reference/programming-model/#providers) for more information.
    /// </summary>
    public partial class Provider : Pulumi.ProviderResource
    {
        /// <summary>
        /// Create a Provider resource with the given unique name, arguments, and options.
        /// </summary>
        ///
        /// <param name="name">The unique name of the resource</param>
        /// <param name="args">The arguments used to populate this resource's properties</param>
        /// <param name="options">A bag of options that control this resource's behavior</param>
        public Provider(string name, ProviderArgs? args = null, ResourceOptions? options = null)
            : base("command", name, args ?? ResourceArgs.Empty, MakeResourceOptions(options, ""))
        {
        }

        private static ResourceOptions MakeResourceOptions(ResourceOptions? options, Input<string>? id)
        {
            var defaultOptions = new ResourceOptions
            {
                Version = Utilities.Version,
            };
            var merged = ResourceOptions.Merge(defaultOptions, options);
            // Override the ID if one was specified for consistency with other language SDKs.
            merged.Id = id ?? merged.Id;
            return merged;
        }
    }

    public sealed class ProviderArgs : Pulumi.ResourceArgs
    {
        public ProviderArgs()
        {
        }
    }
}
